eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentEvents.scala

```diff
@@ -232,6 +235,31 @@ object PaymentFailure {
 
 }
 
+case class PaymentFailureSummary(amount: MilliSatoshi, route: Seq[PublicKey], message: String)
```

Comment:
After rebase, I suggest we move this to the `json` serializers as `PaymentFailureJson`.

Reply:
I moved it to the `json` serializers but didn't rename it `PaymentFailureJson` and didn't provide an implicit conversion, because in most cases we don't want it (we want the websocket and API to return a "full" `PaymentFailure`).

Reply:
I'm not a fan: in my mind the classes in `JsonSerializers` are serialization details and should be `private`. Now we are in-between. Also we don't have to add the serializers to the `JsonSerializers.formats` if we don't want it in the general case.

This is what I mean: 8a9298d1a56634d355d39f5389f63d3b3e6f233d. Also I would have gone with a slighly different format for failure summary: 4bab67ebdafa6a92bcc984178ce8f836233a583f.

Reply:
Ok, I didn't understand what you previously meant, I thought you didn't want to have any `case class` in the `PaymentEvents` file anymore. I'll apply what you suggest.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1966/commits/eb9f63309a905c81e3ef31d30b76302495af0d72, but I didn't include all the additional data you wanted in `PaymentFailure`, I'm really not sure it's useful. We'll see when actually using these logs, we can add later if necessary.